### PR TITLE
Allow DateTimeImmutable instances to be provided as entry, deletion and feed dates

### PR DESCRIPTION
**Pull request by gsteel**
The writers throw an exception if they receive a DateTimeImmutable instance, for example to WriterAbstract::setDateCreated()

Provide a narrative description of what you are trying to accomplish:

- [x] Are you fixing a bug?
  - [x] Detail how the bug is invoked currently.

```php
$myDate = new DateTimeImmutable('@' . 1234567890);
$writer = new Writer\Feed;
$writer->setDateModified($myDate);
```
  - [x] Detail the original, incorrect behaviour.

```
uncaught Zend\Feed\Exception\InvalidArgumentException('Invalid DateTime object or UNIX Timestamp  passed as parameter')
```
  - [x] Detail the new, expected behaviour.

General happiness

  - [x] Base your feature on the `master` branch, and submit against that branch.
  - [x] Add a regression test that demonstrates the bug, and proves the fix.
  - [x] Add a `CHANGELOG.md` entry for the fix.

### CODE REVIEW

File: test/Writer/FeedTest.php [hunk 1]
```diff
@@ -180,6 +181,14 @@ public function testSetDateCreatedUsesDateTimeObject()
         $this->assertEquals($myDate, $writer->getDateCreated());
     }
 
+    public function testSetDateCreatedUsesDateTimeImmutableObject()
+    {
+        $myDate = new DateTimeImmutable('@' . 1234567890);
```
Comment (froschdesign) [inline test/Writer/FeedTest.php:186]: We should definitely use data-providers for the entire test class – _later_!

Comment (gsteel) [reply test/Writer/FeedTest.php:186]: Hell yeah!

**Comment by froschdesign** [comment]
@gsteel 
It would be great if you could also update the documentation. Please have a look at: https://github.com/zendframework/zend-feed/blob/master/docs/book/writer.md#feed-api-methods

Thanks in advance!

**Comment by froschdesign** [comment]
@gsteel 
Please use the `develop` branch as your changes are an improvement and not a bug fix. Thanks!

**Comment by weierophinney** [comment]
Thanks, @gsteel!

**Pull request merged**